heka/sandbox/filters/dollars.lua

```diff
@@ -0,0 +1,268 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+-- Run tests using `lua -e "TestMode = true;" dollars.lua`
+
+--[[
+Monitor submission sizes and counts by channel and document type. This tracks
+factors relevant to budget planning. See Bug 1179751.
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [PipelineBudget]
+    type = "SandboxFilter"
+    filename = "lua_filters/dollars.lua"
+    message_matcher = "Type == 'telemetry'"
+    ticker_interval = 60
+    preserve_data = false
+
+--]]
+
+require "os"
```

Comment:
Most of the functions don't need to be at global scope, same for the bdate_grammar